src/crewai/utilities/fileHandler.py

```diff
@@ -16,5 +17,78 @@ def __init__(self, file_path):
     def log(self, **kwargs):
         now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
         message = f"{now}: ".join([f"{key}={value}" for key, value in kwargs.items()])
-        with open(self._path, "a", encoding = 'utf-8') as file:
+        with open(self._path, "a", encoding="utf-8") as file:
             file.write(message + "\n")
+
+
+class PickleHandler:
```

Comment:
If I got it right, this has started as a generic handler/helper for dealing with pickle marshaling, right!? However, I believe it got really tied up to the training concept – we have a lot of train-related functions. Should we maybe move this to a training module maybe, renaming it to something like `TrainingHandler` or something like that?

Reply:
I see your point, makes sense. I could have the PickleHandler as a utility file handler class. 
And add a TrainingHandler that can Inherit the class and add the `save_trained_data/append` methods, what do you think?

Reply:
That would be perfect! Loved the idea!